timelock-impl/src/main/java/com/palantir/atlasdb/timelock/SecureTimelockServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.meta.When;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.atlasdb.timelock.lock.AsyncResult;
+import com.palantir.lock.v2.LockImmutableTimestampRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.WaitForLocksRequest;
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.tokens.auth.AuthHeader;
+
+public class SecureTimelockServiceImpl implements SecureTimelockService {
```

Comment:
Is there anything in the Timelock service which uses this API?  Things like metrics for timestamp numbers etc?  We need to think how they authorise themselves.